Java/Classes and Methods/Constructors and This keyword/constructor_and_this_keyword.ipynb

```diff
@@ -0,0 +1,347 @@
+        "id": "VDlWeF-QLrjY"
+      },
+      "source": [
+        "So, when we use the new keyword to create an instance of a class the constructor of the class gets invoked. Basically, the Constructor is used to initialise fields of the objects. \n",
+        "Initialize the values of the fields with what exactly you may ask. Hold that thought we will come back to that. \n",
+        "\n",
+        "\n",
+        "---\n",
+        "\n",
+        "##### **RETURN VALUE OF CONSTRUCTOR** \n",
+        "\n",
+        "Does the have a return value? Hmmm trick question. Yes and no, why because even if we don’t explicitly define a return type and do not return any value from the constructor. Java runtime implicitly return the class object when a constructor is called. "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "TU21swD0MWIv"
+      },
+      "source": [
+        "---\n",
+        "## **TYPES OF CONSTRUCTORS**\n",
+        "---\n",
+        "\n",
+        "There are 2 types of constructors: \n",
+        "1.\t**DEFAULT CONSTRUCTOR:**\n",
+        "\n",
+        "If a class does not have a constructor declared explicitly, then JAVA Compiler provides a Default constructor, you know by default. Default constructor does not have any arguments. Using default constructor, the object fields get initialized with default values such as null,0 etc.\n",
+        "\n",
+        "2.\t**PARAMETERIZED CONSTRUCTOR:**\n",
+        "\n",
+        "If we want to initialise object field with user defined values, then we use the Parameterized constructor. It contains arguments in the constructor declaration. Unlike Default constructor java compiler does not provide with parameterized constructor. It has to be declared explicitly. if we declare a parameterized constructor then Java Compiler will not provide with a default constructor. In this case, we have to declare default constructor as well. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "7s8NdYZ9Mn0y"
+      },
+      "source": [
+        "Example:\n",
+        "Class car {\n",
+        "    private int licenceNo;\n",
+        "\t  private String carName;\n",
+        "\t\n",
+        "  car() {} \t\t\t                          //Default Constructor\n",
+        "\t\n",
+        "  car(int licenceNo,String carName) {}\t\t\t//Parameterized Constructor\n",
+        "\n",
+        "}"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "G7SMKVbyMqav"
+      },
+      "source": [
+        "**COPY CONSTRUCTOR:**\n",
+        "\n",
+        "Now, Java also has a copy constructor. But with a twist. Generally, the copy constructor is a constructor which creates an object by initializing it with an object of the same class, which has been created previously. Java does not provide an explicit copy constructor we need to define it ourselves.\n"
```

Comment:
`Generally, the copy constructor is a constructor...` plag from https://www.tutorialspoint.com/what-are-copy-constructors-in-java. Please change the language